sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -427,8 +424,11 @@ export class MessageSender extends LinkEntity {
           if (!options) {
             options = this._createSenderOptions(Constants.defaultOperationTimeoutInMs);
           }
+
           this._sender = await this._context.namespace.connection.createAwaitableSender(options);
           this.isConnecting = false;
+          checkAborted();
```

Comment:
It's a great comment. In the refactor I did do that (since we don't assign to anything so the caller would have no way to do it).

In this case, however, we make sure that if the user has created a sender that we've finished the assignment to this._sender as well. So they can call close() on their sender to kill the link.

Reply:
Gotcha, and we'd just expect the user to close the top-level client to close the connection even if this was cancelled right? That sounds reasonable to me, just making sure I'm following :smile:

Reply:
Yes, they're 1:1 in the Sender. This won't be the exact same in the receiver but I believe it takes care of it a different way.